### PR TITLE
test(js): Convert PermissionSelection to RTL

### PR DESCRIPTION
**Pull request by evanpurkhiser**

### CODE REVIEW

File: static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx [hunk 1]
```diff
@@ -27,74 +30,45 @@ describe('PermissionSelection', () => {
...
-    expect(wrapper.find('SelectField[key="Release"]')).toBeDefined();
-    expect(wrapper.find('SelectField[key="Event"]')).toBeDefined();
-    expect(wrapper.find('SelectField[key="Organization"]')).toBeDefined();
-    expect(wrapper.find('SelectField[key="Member"]')).toBeDefined();
+    expect(screen.getByRole('textbox', {name: 'Project'})).toBeInTheDocument();
+    expect(screen.getByRole('textbox', {name: 'Team'})).toBeInTheDocument();
+    expect(screen.getByRole('textbox', {name: 'Release'})).toBeInTheDocument();
+    expect(screen.getByRole('textbox', {name: 'Issue & Event'})).toBeInTheDocument();
+    expect(screen.getByRole('textbox', {name: 'Organization'})).toBeInTheDocument();
+    expect(screen.getByRole('textbox', {name: 'Member'})).toBeInTheDocument();
```
Comment (gggritso) [inline static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx:38]: `it.each` would be _beautiful_ here

File: static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx [hunk 2]
```diff
@@ -27,74 +30,45 @@ describe('PermissionSelection', () => {
...
-        Organization: 'read',
-      },
-    });
-
-    expect(instance.permissionStateToList()).toEqual([
-      'project:read',
-      'project:write',
-      'project:releases',
-      'org:read',
-    ]);
```
Comment (gggritso) [inline static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx:79]: This test is _slightly_ lost, but not a big deal. Would be nice to check the API call mock for form submission and check that it gets the right scopes in this correct format

File: static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx [hunk 3]
```diff
@@ -27,74 +30,45 @@ describe('PermissionSelection', () => {
...
-    expect(getStateValue('Release')).toEqual('admin');
-    expect(getStateValue('Event')).toEqual('admin');
-    expect(getStateValue('Organization')).toEqual('read');
-    expect(getStateValue('Member')).toEqual('no-access');
+    expect(model.getValue('Project--permission')).toEqual('write');
+    expect(model.getValue('Team--permission')).toEqual('read');
+    expect(model.getValue('Release--permission')).toEqual('admin');
+    expect(model.getValue('Event--permission')).toEqual('admin');
+    expect(model.getValue('Organization--permission')).toEqual('read');
+    expect(model.getValue('Member--permission')).toEqual('no-access');
```
Comment (gggritso) [inline static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx:72]: IMO you might as well leave the `model` out and assert on the value of the hidden inputs, but that's just me! ✨ integrationy ✨

**Review by gggritso — APPROVED** [review]
🧹

File: static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx (hunk 3 above)
Comment (evanpurkhiser) [reply static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx:72]: It's kind of hard with RTL to check hidden input values actually hah

Comment (gggritso) [reply static/app/views/settings/organizationDeveloperSettings/permissionSelection.spec.jsx:72]: Oh interesting! Alright 👌🏻

**Pull request merged**